pywbem/_logging.py

```diff
@@ -14,371 +14,303 @@
+level.  Also, when either logger is active, each new WBEMConnection is logged.
 
-#: List of log destinations that :class:`~pywbem._logging.PywbemLoggers`
-#: recognizes.
-LOG_DESTINATIONS = ['file', 'stderr', 'none']
+There are three parts to configuration of the pywbem loggers:
 
-#: List of the log detail strings that :class:`~pywbem._logging.PywbemLoggers`
-#: recognizes.
-LOG_DETAIL_LEVELS = ['all', 'min']
+* **Configure the loggers** - This involves setting the python logging
+  parameters that control the logger and its output and may be done by either
+  python logging methods or methods within WBEMConnection. It requires defining
+  the specific loggers that will control pywbem logging.
 
-#: Default log detail level string if none is supplied with a call to the
-#: :class:`pywbem._logging.PywbemLoggers` methods that configure pywbem named
-#: loggers
-DEFAULT_LOG_DETAIL_LEVEL = 'min'
+* **Set the `detail_level` for pywbem log** - Because pywbem logs can
+  generate large quantities of information, pywbem allows the user to control
+  the quantity of information in each log.  Since this is a pywbem feature it
+  can only be configured by the pywbem :class:`~pywbem.WBEMConnection`
+  methods that configure loggers. Generally the detail levels are:
 
-#: Default maximum log entry size if none is supplied when creating
-#: :class:`~pywbem.LogOperationRecorder` objects.
-DEFAULT_MAX_LOG_ENTRY_SIZE = 1000
+  * `all` - All of the information is output.
 
-#: Default log destination if none is defined when named loggers are
-#: configured. 'none' means that there is no logging.
-DEFAULT_LOG_DESTINATION = 'none'
+  * `summary` - Only summary information is output. For api responses this
+    is primarily the count and type of objects returned. For http logs
+    the HTTP header information is output.
 
+  * `paths` - For api responses that include CIMInstanceNames or CIMClassNames,
+    (paths components in the objects) string representation of the
+    path components is output. For responses that  do not include CIMInstance
+    paths the complete objects representations are output.
 
-class MetaPywbemLoggers(type):
-    """
-    **Experimental:** *New in pywbem 0.11 as experimental.*
+  * :class:`py:int` - All of the information is output up to the size defined
+    by the integer value.  This applies to all log entries.
 
-    This metaclass allows the definition of __str__ and __repr__ at the
-    class level in the subclass.
-    """
-    loggers = {}
+* **Activate the loggers** - pywbem separates the configuration of the loggers
+  from activation to allow logger configuration by either pywbem methods or
+  standard python logging methods. Activation sets the pywbem loggers to a
+  state where they will generate log output when any of the WBEMConnection
+  methods that communicates with a WBEM server is called.  This involves
+  definining logging for a single connection. and optionally setting the
+  detail_level for the   log output
```

Comment:
????? I would not understand that as a user that did not look at the code. I think the simple reason for having a concept of activation is that we want to optimize performance in case logging is deactivated.
Also, we should state how activation and deactivation is performed by the user, with a link.